frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,228 @@
+  const podToCreate = React.useMemo(() => {
+    return getDebugPod(debugPodName, podData, podContainerName);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debugPodName, podContainerName]);
+
+  React.useEffect(() => {
+    const deleteDebugPod = async (podToDelete) => {
+      try {
+        await k8sKillByName(PodModel, podToDelete, podNamespace);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('Could not delete container terminal debug pod.', e);
+      }
+    };
+    let newDebugPod;
+    const createDebugPod = async () => {
+      try {
+        // await new Promise((resolve) => setTimeout(resolve, 1000));
+        newDebugPod = await k8sCreate(PodModel, podToCreate);
+        setGeneratedDebugPodName(newDebugPod?.metadata?.name);
+      } catch (e) {
+        setErrorMessage(e.message);
+      }
+    };
+    createDebugPod();
+
+    window.addEventListener('beforeunload', deleteDebugPod);
+    return () => {
+      if (newDebugPod) {
+        deleteDebugPod(newDebugPod.metadata.name);
+      }
+      window.removeEventListener('beforeunload', deleteDebugPod);
+    };
+  }, [debugPodName, podNamespace, podToCreate]);
+
+  const [debugPod, loaded, err] = useK8sWatchResource<PodKind>(
+    generatedDebugPodName
+      ? {
+          isList: false,
+          kind: 'Pod',
+          name: generatedDebugPodName,
+          namespace: podNamespace,
+        }
+      : {},
+  );
+
+  if (errorMessage) {
+    return <DebugTerminalError error={errorMessage} />;
+  }
+
+  if (generatedDebugPodName) {
+    if (err) {
+      return <DebugTerminalError error={err} />;
+    }
+    if (loaded) {
+      return <DebugTerminalInner activeContainer={containerName} debugPod={debugPod} />;
+    }
+  }
+
+  return <LoadingBox />;
+};
+
+export const DebugTerminalPage: React.FC<DebugTerminalPageProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const {
+    params: { podName, ns, name },
+    url,
+  } = match;
+
+  const [podData, loaded, err] = useK8sWatchResource<PodKind>({
+    isList: false,
+    kind: 'Pod',
+    name: podName,
+    namespace: ns,
+  });
+
+  return (
+    <div>
+      <Helmet>
+        <title>{t('public~Debug {{name}}', { name })}</title>
+      </Helmet>
+      <PageHeading
+        detail={true}
+        title={`${t('public~Debug {{name}}', { name })}`}
+        kind="Container"
+        breadcrumbs={[
+          { name: t('public~Pods'), path: getBreadcrumbPath(match, 'pods') },
+          {
+            name: podName,
+            path: resourcePath('Pod', podName, ns),
+          },
+          {
+            name: t('public~Container details'),
+            path: `${resourcePath('Pod', podName, ns)}/containers/${name}`,
+          },
+          { name: t('public~Debug container'), path: url },
+        ]}
+      />
+      {loaded && <DebugTerminal podData={podData} containerName={name} />}
+      {err && <LoadingBox message={err} />}
```

Comment:
`loaded` and `err` can both be true, right? Do you mean to show the terminal if `err` is true?

Reply:
I guess so....?  I will add a check to make sure there is no err.